src/Discovery/Discovery.tsx

```diff
@@ -348,7 +359,7 @@ const Discovery: React.FunctionComponent<Props> = (props: Props) => {
       // Reinitialize search
       props.onSearchChange(props.searchTerm);
       // Cache only the Full Text Search object
-      if(selectedSearchableTextFields.length === 0 ) {
+      if (selectedSearchableTextFields.length === allSearchableFields.length) {
```

Comment:
Will this still only cache the Full Text Search object? If not, the comment on line 361 should be updated.